### PR TITLE
HDXDSYS-894 sector, org_type and get_code_from_name improvements

### PR DESCRIPTION
**Pull request by mcarans**
- Normalise keys in sector and org type lookups on creation
- Fold unofficial mappings into self.data in sector and org type
- Simplify get_code_from_name

### CODE REVIEW

File: src/hapi/pipelines/utilities/mappings.py [hunk 1]
```diff
@@ -9,47 +9,40 @@
...
         fuzzy_match (bool): Allow fuzzy matching or not
 
     Returns:
         str or None: matching code
-        str: clean name
-        bool: whether to add the mapping to the unofficial mappings dictionary
     """
     code = code_lookup.get(name)
     if code:
         return code, name, False
```
Comment (turnerm) [inline src/hapi/pipelines/utilities/mappings.py:27]: This should just be `code` I thnk

File: src/hapi/pipelines/database/sector.py [hunk 2]
```diff
@@ -23,17 +24,16 @@ def __init__(
...
 
         def parse_sector_values(code: str, name: str):
             if code != "intersectoral":
-                self.data[name] = code
-                self.data[code] = code
+                self.data[normalise(name)] = code
+                self.data[normalise(code)] = code
                 pattern = code.lower().replace("-", "_")
                 pattern = TagPattern.parse(f"#*+{pattern}")
                 self.pattern_to_code[pattern] = code
```
Comment (turnerm) [inline src/hapi/pipelines/database/sector.py:39]: Are these three lines needed anymore?

**Review by turnerm — APPROVED** [review]
Really nice simplifications!

File: src/hapi/pipelines/utilities/mappings.py (hunk 1 above)
Comment (mcarans) [reply src/hapi/pipelines/utilities/mappings.py:27]: You are right, thanks

File: src/hapi/pipelines/database/sector.py (hunk 2 above)
Comment (mcarans) [reply src/hapi/pipelines/database/sector.py:39]: Good spot - pattern_to_code is not used anywhere

**Comment by b-j-mills** [comment]
Looks great, much easier to understand!  Have you checked if changing any characters in the lookup causes the tests to hang?

**Comment by mcarans** [comment]
> Looks great, much easier to understand! Have you checked if changing any characters in the lookup causes the tests to hang?

Good idea! I checked by deleting some sector and org type mappings and they don't hang.

**Pull request merged**